tests/effect_test.cpp

```diff
@@ -260,13 +279,119 @@ TEST_CASE( "effect decay", "[effect][decay]" )
         CHECK( to_turns<int>( eff_debugged.get_duration() ) == 2 );
     }
 
-    // TODO:
-    // When intensity > 1
-    // - and duration < max_duration
-    // - and int_decay_tick is set (from effect JSON)
-    // - and time % decay_tick == 0
-    // Then:
-    // - add int_decay_step to intensity (default -1)
+    SECTION( "intensity decay triggers on the appropriate turns" ) {
+        effect eff_debugged( effect_source::empty(), &eff_id.obj(), 1_hours, bodypart_str_id( "bp_null" ),
+                             false, 10, calendar::turn );
+        // Ensure it has a decay tick  of 2 turns and a decay step of -1, and int removal is allwed
+        // Also check max duration
+        REQUIRE( eff_debugged.get_max_duration() == 1_hours );
+        REQUIRE( eff_debugged.get_intensity() == 10 );
+        REQUIRE( eff_debugged.get_int_decay_step() == -1 );
+        REQUIRE( eff_debugged.get_int_decay_tick() == 2 );
+        REQUIRE( eff_debugged.get_int_decay_remove() == true );
+        // Reset time
+        calendar::turn = calendar::start_of_cataclysm;
+
+        // First decay - at max duration, no intensity decay
+        CHECK( eff_debugged.get_duration() == eff_debugged.get_max_duration() );
+        eff_debugged.decay( rem_ids, rem_bps, calendar::turn, false );
+        CHECK( eff_debugged.get_intensity() == 10 );
+        // Effect not removed
+        CHECK( rem_ids.empty() );
+        CHECK( rem_bps.empty() );
+
+        // Progress turns, checking for the appropriate intensity and no removal
+
+        // Turn 1, no intensity change
+        calendar::turn += 1_turns;
+        eff_debugged.decay( rem_ids, rem_bps, calendar::turn, false );
+        CHECK( eff_debugged.get_intensity() == 10 );
+        // Effect not removed
+        CHECK( rem_ids.empty() );
+        CHECK( rem_bps.empty() );
+
+        // Turn 2, intensity decays
+        calendar::turn += 1_turns;
+        eff_debugged.decay( rem_ids, rem_bps, calendar::turn, false );
+        CHECK( eff_debugged.get_intensity() == 9 );
+        // Effect not removed
+        CHECK( rem_ids.empty() );
+        CHECK( rem_bps.empty() );
+
+        // Turn 3, no intensity change
+        calendar::turn += 1_turns;
+        eff_debugged.decay( rem_ids, rem_bps, calendar::turn, false );
+        CHECK( eff_debugged.get_intensity() == 9 );
+        // Effect not removed
+        CHECK( rem_ids.empty() );
+        CHECK( rem_bps.empty() );
+
+        // Skip the intermediate levels, set intensity to 1 manually before proceeding
+        eff_debugged.set_intensity( 1 );
+        calendar::turn += 1_turns;
+        eff_debugged.decay( rem_ids, rem_bps, calendar::turn, false );
+        CHECK( eff_debugged.get_intensity() == 0 );
+        // Effect is removed
+        CHECK( rem_ids.size() == 1 );
+        CHECK( rem_bps.size() == 1 );
```

Comment:
```suggestion
        REQUIRE( rem_ids.size() == 1 );
        REQUIRE( rem_bps.size() == 1 );
```
Since the remainder of this test doesn't work if these aren't true.